mono/metadata/metadata.c

```diff
@@ -5014,6 +5012,7 @@ mono_metadata_fnptr_equal (MonoMethodSignature *s1, MonoMethodSignature *s2, gbo
  * mono_metadata_type_equal:
  * @t1: a type
  * @t2: another type
+ * @signature_only: If true, treat generics which are instantiated separately but "shaped the same" as equal
```

Comment:
Shaped the same is ambiguous. It's simple, they are treated just for their positional values instead of container + position.